test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -745,26 +756,46 @@ protected void deleteAllNodeShutdownMetadata() throws IOException {
         }
     }
 
+    protected void wipeSystemResources() throws IOException {
+        // TODO: 7.16 for backwards compatibility
+        if (minimumNodeVersion().onOrAfter(Version.CURRENT)) {
+            // feature reset deletes system indices
+            final Request postRequest = new Request("POST", "/_features/_reset");
+            Response response;
+            try {
+                response = adminClient().performRequest(postRequest);
+            } catch (ResponseException e) {
+                response = e.getResponse();
+        }
+        if (response.getStatusLine().getStatusCode() != 200) {
+            // TODO: since there is a bug in transform reset (link to issue!) we can't throw an error here
+            // throw new IllegalStateException("Failed to reset all feature states: "
+            logger.warn("Failed to reset all feature states: "
+                + Streams.copyToString(new InputStreamReader(response.getEntity().getContent(), StandardCharsets.UTF_8)));
+            }
```

Comment:
I am hoping this isn't the case any longer :). There were bugs at the start, but if there are some current ones, it would be good to know what they are.

Reply:
I kept running into errors like this one (note that there are some investigation log lines in that output): https://gradle-enterprise.elastic.co/s/rdu2sojvv7njq/console-log?task=:x-pack:plugin:transform:qa:single-node-tests:javaRestTest

You should be able to reproduce it at commit e754de22529 with:
```
➜  ./gradlew ':x-pack:plugin:transform:qa:single-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.TransformGetAndGetStatsIT.testGetStatsWithContinuous" -Dtests.locale=ro-RO -Dtests.timezone=Africa/Asmera -Druntime.java=11
```

I wasn't able to pin down exactly where the problem was, so I was going to take another stab at it before writing up an issue. At one point I did have a logger giving me the stack trace for the failure, for example here: https://gradle-enterprise.elastic.co/s/psveo5xaja3mc/console-log?task=:x-pack:plugin:transform:qa:single-node-tests:javaRestTest

Pasted below for your convenience...

<details><summary>big stack trace</summary>

```
» [2021-08-18T14:19:27,042][WARN ][r.suppressed             ] [javaRestTest-0] path: /_features/_reset, params: {}    
»  org.elasticsearch.ElasticsearchStatusException: Failed node [X0DmH_ZZSX-GFG-9I2ZcdQ]    
»      at org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:342) ~[?:?]    
»      at org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:317) ~[?:?]    
»      at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:134) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:300) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onFailure(TransportTasksAction.java:287) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleException(TransportTasksAction.java:263) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService$5.handleException(TransportService.java:629) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1174) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1283) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1257) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:50) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.onFailure(SecurityServerTransportInterceptor.java:198) ~[?:?]    
»      at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$AbstractFilterListener.onFailure(SecurityServerTransportInterceptor.java:292) ~[?:?]    
»      at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeSystemUser(AuthorizationService.java:452) ~[?:?]    
»      at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(AuthorizationService.java:226) ~[?:?]    
»      at org.elasticsearch.xpack.security.transport.ServerTransportFilter.lambda$inbound$1(ServerTransportFilter.java:109) ~[?:?]    
»      at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:134) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:342) ~[?:?]    
»      at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:184) ~[?:?]    
»      at org.elasticsearch.xpack.security.transport.ServerTransportFilter.inbound(ServerTransportFilter.java:100) ~[?:?]    
»      at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:275) ~[?:?]    
»      at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:61) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService.sendLocalRequest(TransportService.java:792) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService$3.sendRequest(TransportService.java:114) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:736) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor.sendWithUser(SecurityServerTransportInterceptor.java:126) ~[?:?]    
»      at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$1.sendRequest(SecurityServerTransportInterceptor.java:109) ~[?:?]    
»      at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:650) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:576) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.start(TransportTasksAction.java:249) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.support.tasks.TransportTasksAction.doExecute(TransportTasksAction.java:83) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$doExecute$0(TransportStopTransformAction.java:188) ~[?:?]    
»      at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:134) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.xpack.transform.persistence.IndexBasedTransformConfigManager.lambda$expandTransformIds$10(IndexBasedTransformConfigManager.java:466) ~[?:?]    
»      at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:134) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:31) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.client.node.NodeClient.lambda$executeLocally$0(NodeClient.java:103) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:169) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:163) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:31) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.lambda$applyInternal$2(SecurityActionFilter.java:146) ~[?:?]    
»      at org.elasticsearch.action.ActionListener$DelegatingFailureActionListener.onResponse(ActionListener.java:217) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.ActionListener$RunAfterActionListener.onResponse(ActionListener.java:339) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.AbstractSearchAsyncAction.sendSearchResponse(AbstractSearchAsyncAction.java:655) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.ExpandSearchPhase.run(ExpandSearchPhase.java:109) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:424) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:418) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:219) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$1(FetchSearchPhase.java:101) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:107) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:84) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:737) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]    
»      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]    
»      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]    
»      at java.lang.Thread.run(Thread.java:829) [?:?]

```
</details>